core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterDMTest.java

```diff
@@ -3386,6 +3386,20 @@ private void checkHavingAliasSameAsColumn(boolean upperAlias) {
     assertThat(toSql(root, DatabaseProduct.BIG_QUERY.getDialect()), isLinux(expectedBiqQuery));
   }
 
+  @Test public void testConcat2FunctionWithBooleanArgumentRelToSql() {
+    final RelBuilder builder = relBuilder();
+    final RexNode concatRexNode =
+        builder.call(SqlLibraryOperators.CONCAT2, builder.literal("foo"),
+            builder.literal("bar"), builder.literal("true"));
+    final RelNode root = builder
+        .scan("EMP")
+        .project(builder.alias(concatRexNode, "CR"))
+        .build();
+    final String expectedSql = "SELECT CONCAT('foo', 'bar') AS \"CR\"\n"
```

Comment:
we could add an assertion for sqlNode unparsing as well

```
final String expectedSql = "SELECT CONCAT('foo', 'bar', 'true') AS \"CR\"\n"
        + "FROM \"scott\".\"EMP\"";
    final String expectedPostgresSql = "SELECT CONCAT('foo', 'bar') AS \"CR\"\n"
        + "FROM \"scott\".\"EMP\"";
    assertThat(toSql(root, DatabaseProduct.CALCITE.getDialect()), isLinux(expectedSql));
    assertThat(toSql(root, DatabaseProduct.POSTGRESQL.getDialect()), isLinux(expectedPostgresSql));
```

Reply:
added